pkg/remoteconfig/state/testingutils_test.go

```diff
@@ -188,23 +188,19 @@ func newAPMSamplingFile() apmsampling.APMSampling {
 		}},
 	}
 
-	return apmConfig
+	raw, _ := apmConfig.MarshalMsg(nil)
+
+	return raw
 }
 
-func addAPMSamplingFile(id string, version int64, file apmsampling.APMSampling, targets *data.Targets) (string, data.Hashes, []byte) {
+func addAPMSamplingFile(id string, version int64, file []byte, targets *data.Targets) (string, data.Hashes, []byte) {
 	path := fmt.Sprintf("datadog/3/%s/%s/config", ProductAPMSampling, id)
 
-	buf := make([]byte, 0, file.Msgsize())
-	out, err := file.MarshalMsg(buf)
-	if err != nil {
-		panic(err)
-	}
-
-	tfm := generateRCTargetFileMeta(out, version)
+	tfm := generateRCTargetFileMeta(file, version)
 
 	targets.Targets[path] = tfm
 
-	return path, tfm.Hashes, out
+	return path, tfm.Hashes, file
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Is `file` modified by `generateRCTargetFileMeta` (kind of an antipattern if that's the case)? If not, then you probably don't need to return it

Reply:
oh right good catch, fixed :)